lib/thrift/generator/client.ex

```diff
@@ -0,0 +1,270 @@
+defmodule Thrift.Generator.ClientGenerator do
+  defmodule Framed do
+    alias Thrift.Generator.Service
+    alias Thrift.Parser.Models.Function
+
+    def generate(service_module, service) do
+      functions = service.functions
+      |> Map.values
+      |> Enum.map(&generate_function(service_module, &1))
+
+      quote do
+        defmodule Client.Framed do
+          alias Thrift.Protocols.Binary, as: BinaryBase
+          alias Thrift.Exceptions.TApplicationException
+
+          defmodule State do
+            defstruct host: nil, port: nil, tcp_opts: nil, timeout: 5000, sock: nil
+          end
+          use Connection
+
+          def init({host, port, tcp_opts, timeout}) do
+            s = %State{host: host,
+                       port: port,
+                       tcp_opts: tcp_opts,
+                       timeout: timeout,
+                       sock: nil}
+
+            {:connect, :init, s}
+          end
+
+          def start_link(host, port, tcp_opts, timeout \\ 5000) do
+            Connection.start_link(__MODULE__, {host, port, tcp_opts, timeout})
+          end
+
+          def send(conn, data), do: Connection.call(conn, {:send, data})
+
+          def recv(conn, bytes, timeout \\ 5000) do
+            Connection.call(conn, {:recv, bytes, timeout})
+          end
+
+          def close(conn), do: Connection.call(conn, :close)
+
+          def connect(_, %{sock: nil,
+                           host: host,
+                           port: port,
+                           tcp_opts: opts,
+                           timeout: timeout}=s) do
+
+            case :gen_tcp.connect(host, port, [active: false, packet: 4] ++ opts, timeout) do
+              {:ok, sock} ->
+                {:ok, %{s | sock: sock}}
+
+              {:error, _} ->
+                {:backoff, 1000, s}
+            end
+          end
+
+          def disconnect(info, %{sock: sock} = s) do
+            :ok = :gen_tcp.close(sock)
+            case info do
+              {:close, from} ->
+                Connection.reply(from, :ok)
+
+              {:error, :closed} ->
+                :error_logger.format("Connection closed~n", [])
```

Comment:
Its in the connection example to show `:inet.format_error` and to be clear that need to log. The example predates Logger I think. Logger is designed to use a custom backend because don't want to "force" one way of handling files.